src/ape/managers/config.py

```diff
@@ -183,6 +184,41 @@ def serialize(self) -> Dict:
 
         return project_config
 
+    @contextmanager
+    def using_project(self, project_folder: Path, contracts_folder: Path) -> Generator:
+        """
+        Temporarily change the project context.
+
+        Usage example::
+
+            from pathlib import Path
+            from ape import config, Project
+
+            project_path = Path("path/to/project")
+            contracts_path = project_path / "contracts"
+
+            with config.using_project(project_path, contracts_path):
+                my_project = Project(project_path)
+
+        Args:
+            project_folder (pathlib.Path): The path of the context's project.
+            contracts_folder (pathlib.Path): The path to the context's source files.
```

Comment:
Is this ever not using something that can be set by the config in `project_folder`?

Reply:
ah, so there is a contracts folder outside of the project?

Reply:
The whole point of this method is to not use the project config one

Reply:
This is how I am using it in the dependencies compilation:

```python
            with self._compilers.config.using_project(
                temp_project_path, contracts_folder=temp_contracts_path
            ):
                contract_types = self._compilers.compile(sources)


```